### PR TITLE
Feature/80 secondary navigation

### PR DESCRIPTION
**Pull request by Pete-Robelou**
- Desktop = Horizontal list
- Tablet/phone = Vertical list
- Tablet/phone (JS enhanced) = expandable dropdown menu

### CODE REVIEW

File: src/css/component.nav-secondary.css [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    position: absolute;
+    top: var(--size-11);
+    width: var(--size-full);
+  }
+
+  .js-nav-secondary--opened .nav-secondary__sub {
+    display: block;
+  }
+
+  .js-nav-secondary .nav-secondary__sub a {
```
Comment (adamduncan) [inline src/css/component.nav-secondary.css:122]: Thought: I wonder if these could be `display: block` to increase the expected clickable area for mobile devices?

File: src/_includes/components/navigation-secondary.njk [hunk 2]
```diff
@@ -1,39 +1,25 @@
...
+      {%- for item in primaryNav -%}
+        {%- if page.url | startsWith(item.url) -%}
+          <li class="nav-secondary__wrapper"><a class="nav-secondary__top" href="{{ item.url | url }}" data-nav-secondary-toggle>{{ item.title }}</a>
           {%- if item.children %}
-            {%- for subitem in item.children %}
-              <li>
-                <a href="{{ subitem.url | url }}" {% if page.url | startsWith(subitem.url) %} aria-current="page"{% endif %}>{{ subitem.title }}</a>
-              </li>
-            {%- endfor %}
+            <span class="nav-secondary__pipe">/</span>
```
Comment (adamduncan) [inline src/_includes/components/navigation-secondary.njk:12]: Thought: Could this be done in style and avoid the need to include it in code/hide it for small screens/remove semantically?

File: src/css/component.nav-secondary.css [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  line-height: var(--size-5);
+}
+
+.nav-secondary__sub {
+  font-size: var(--font-size-sm);
+  line-height: var(--size-5);
+  margin-bottom: 0;
+}
+
+.js-nav-secondary {
```
Comment (adamduncan) [inline src/css/component.nav-secondary.css:54]: Option: Might this be `.js .nav-secondary` and leverage the document's `.js` class, as an alternative to creating a new set of `.js-`-prefixed classes?

File: src/js/nav-secondary.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  if (navigation) {
+    const toggle = navigation.querySelector('[data-nav-secondary-toggle]');
+    const navigation_sub = navigation.querySelector('[data-nav-secondary-sub]');
+    const media_query = window.matchMedia('(width < 64em)')
+  
+    navigation.classList.add('js-nav-secondary');
+  
+    function toggle_nav(e) {
+      if (navigation.classList.contains("js-nav-secondary--opened")) {
+        navigation.classList.remove('js-nav-secondary--opened');
```
Comment (adamduncan) [inline src/js/nav-secondary.js:16]: Wonder if we might be able to use some `aria-pressed`/`aria-expanded` semantics here to manage button state. E.g. .https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#associated_aria_roles_states_and_properties

Might also mean we add/remove `role="button"` from control at different breakpoints?

File: src/js/nav-secondary.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      }
+      else {
+        toggle.removeEventListener('click', toggle_nav);
+        navigation_sub.setAttribute("aria-hidden", "false");
+        navigation.classList.remove('js-nav-secondary--opened');
+      }
+    }
+    
+    // Register event listener
+    media_query.addEventListener('change', handle_size_change);
```
Comment (adamduncan) [inline src/js/nav-secondary.js:38]: Question: Do we need to to do the `addEventListener`/`addListener` dance here for older Safari?

**Review by adamduncan — APPROVED** [review]
Looks really nice, @Pete-Robelou. A couple of thoughts. No showstoppers to merge.

File: src/js/nav-secondary.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+"use strict";
+/*global document window*/
+
+(function() {
+  const navigation = document.querySelector('[data-nav-secondary');
```
Comment (adamduncan) [inline src/js/nav-secondary.js:5]:
```suggestion
  const navigation = document.querySelector('[data-nav-secondary]');
```

Minor: Missing closing `]` data attr selector

File: src/css/component.nav-secondary.css (hunk 1 above)
Comment (Pete-Robelou) [reply src/css/component.nav-secondary.css:122]: I've made them width full to achieve the same thing.

Apples vs Oranges 🙂

File: src/_includes/components/navigation-secondary.njk (hunk 2 above)
Comment (Pete-Robelou) [reply src/_includes/components/navigation-secondary.njk:12]: It certainly can, all done 👍

File: src/js/nav-secondary.js (hunk 6 above)
Comment (Pete-Robelou) [reply src/js/nav-secondary.js:5]: Fixed.

File: src/css/component.nav-secondary.css (hunk 3 above)
Comment (Pete-Robelou) [reply src/css/component.nav-secondary.css:54]: Yeah, good shout. All done.

File: src/js/nav-secondary.js (hunk 4 above)
Comment (Pete-Robelou) [reply src/js/nav-secondary.js:16]: Yeah, nice. I've switched over to `aria-pressed/aria-expanded` and added the `role="button"` when needed. Just needed a minor adjustment to the CSS to pull it all together.

File: src/js/nav-secondary.js (hunk 5 above)
Comment (Pete-Robelou) [reply src/js/nav-secondary.js:38]: Probably. I'll pick this one up in the refactor JS ticket.

**Pull request merged**